### PR TITLE
feat: Added issue comment stream

### PR DESCRIPTION
**Pull request by prakharcode**
Added a new stream, fixed schema of bunch of fields in issue stream. This PR has more commit as getting the head of my main level

### CODE REVIEW

**Comment by prakharcode** [comment]
@edgarrmondragon I'm facing an issue since yesterday that the replication based on `updated` key is not working for IssueStream. Changing it to `id` attributes fixes the problem but could be a problem for state mangement.
Can you replicate this on your end?

File: tap_jira/streams.py [hunk 1]
```diff
@@ -66,30 +68,6 @@ class UsersStream(JiraStream):
...
-
-        params: dict = {}
-        if next_page_token:
-            params["startAt"] = next_page_token
-        if self.replication_key:
-            params["sort"] = "asc"
-            params["order_by"] = self.replication_key
-
-        return params
-
```
Comment (edgarrmondragon) [inline tap_jira/streams.py:92]: Was this removed because it's redundant with the parent class?

File: tap_jira/streams.py [hunk 2]
```diff
@@ -407,13 +385,334 @@ class IssueStream(JiraStream):
     """
 
     name = "issues"
-    path = "/search?maxResults=1"
+    path = "/search?maxResults=10"
     primary_keys = ["id"]
-    replication_key = "updated"
+    replication_key = "id"
```
Comment (edgarrmondragon) [inline tap_jira/streams.py:390]:
```suggestion
    replication_key = "id"
```

Does this mean users will not get new records when an issue is updated?

File: tap_jira/streams.py [hunk 3]
```diff
@@ -4279,14 +3570,21 @@ class IssueChangeLogStream(JiraStream):
 
     path = "/issue/{issue_id}/changelog"
 
+    replication_method = "INCREMENTAL"
+
```
Comment (edgarrmondragon) [inline tap_jira/streams.py:3574]: This is redundant since you're declaring `replication_key`

```suggestion
```

File: tap_jira/streams.py [hunk 4]
```diff
@@ -3019,18 +2310,9 @@ def get_url_params(
...
-        try:
-            row["created"] = row.get("fields").get("created")
-            row["updated"] = row.get("fields").get("updated")
-        except:
-            pass
-
-        return super().post_process(row, context)
+    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
+        """Return a context dictionary for child streams."""
+        return {"issue_id": record["id"]}
```
Comment (edgarrmondragon) [inline tap_jira/streams.py:2315]: This should fix #43 👍

File: tap_jira/streams.py [hunk 5]
```diff
@@ -3645,79 +2926,84 @@ class SprintStream(JiraStream):
...
+    replication_key = datetime keys for replication
+    records_jsonpath = json response body
+    """
 
-                sprint = Sprint(self._tap, schema={"properties": {}})
+    name = "sprints"
+    parent_stream_type = BoardStream
+    path = "/board/{board_id}/sprint?maxResults=100"
+    replication_method = "INCREMENTAL"
+    replication_key = "id"
```
Comment (edgarrmondragon) [inline tap_jira/streams.py:2972]: Ditto, `id` only makes sense as a replication key if records are immutable (e.g. events)

File: tap_jira/streams.py (hunk 1 above)
Comment (prakharcode) [reply tap_jira/streams.py:92]: yes

File: tap_jira/streams.py (hunk 2 above)
Comment (prakharcode) [reply tap_jira/streams.py:390]: I have added a `get_url_params` function that fetches any record that is updated within the start/end data

File: tap_jira/streams.py (hunk 3 above)
Comment (prakharcode) [reply tap_jira/streams.py:3574]: okay, thanks for this. Will be adding a new stream and then push this too.

**Review by edgarrmondragon — APPROVED** [review]
Thanks @prakharcode!

**Pull request merged**